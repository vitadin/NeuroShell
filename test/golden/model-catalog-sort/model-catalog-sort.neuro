%% Sort options test
\set[_echo_commands="true"]
%% Tests sorting models by name and provider
\model-catalog[sort=name]
\model-catalog[sort=provider]