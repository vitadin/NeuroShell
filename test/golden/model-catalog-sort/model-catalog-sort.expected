Setting _echo_commands = true
%%> \model-catalog[sort="name"]
Model Catalog (6 models):
  [CO37] Claude 3.7 Opus (claude-3-7-opus-20240229)
    Provider: anthropic
    Context: 200,000 tokens (max output: 4,096 tokens)
    Capabilities: text, reasoning, analysis, coding, writing, math, research, visual-reasoning
    Modalities: text-input, text-output, image-input
    Pricing: $15.00/1M input, $75.00/1M output tokens
    Features: streaming, function-calling, structured-outputs, vision
    Knowledge cutoff: 2023-08-01
    Description: Top-level intelligence, fluency, and understanding
  [CS37] Claude 3.7 Sonnet (claude-3-7-sonnet-20250219)
    Provider: anthropic
    Context: 200,000 tokens (max output: 64,000 tokens)
    Capabilities: text, reasoning, analysis, coding, writing, math, visual-reasoning, extended-thinking
    Modalities: text-input, text-output, image-input
    Pricing: $3.00/1M input, $15.00/1M output tokens
    Features: streaming, function-calling, structured-outputs, vision
    Knowledge cutoff: 2024-11-01
    Description: High intelligence with toggleable extended thinking
  [CO4] Claude Opus 4 (claude-opus-4-20250514)
    Provider: anthropic
    Context: 200,000 tokens (max output: 32,000 tokens)
    Capabilities: text, reasoning, analysis, coding, writing, math, research, visual-reasoning, extended-thinking
    Modalities: text-input, text-output, image-input
    Pricing: $15.00/1M input, $75.00/1M output tokens
    Features: streaming, function-calling, structured-outputs, vision
    Knowledge cutoff: 2025-03-01
    Description: Highest level of intelligence and capability
  [CS4] Claude Sonnet 4 (claude-sonnet-4-20250514)
    Provider: anthropic
    Context: 200,000 tokens (max output: 64,000 tokens)
    Capabilities: text, reasoning, analysis, coding, writing, math, visual-reasoning, extended-thinking
    Modalities: text-input, text-output, image-input
    Pricing: $3.00/1M input, $15.00/1M output tokens
    Features: streaming, function-calling, structured-outputs, vision
    Knowledge cutoff: 2025-03-01
    Description: High intelligence and balanced performance
  [O3] O3 (o3)
    Provider: openai
    Context: 200,000 tokens (max output: 100,000 tokens)
    Capabilities: text, reasoning, analysis, coding, writing, math, visual-reasoning, instruction-following, technical-writing
    Modalities: text-input, text-output, image-input
    Pricing: $2.00/1M input, $8.00/1M output tokens
    Features: streaming, function-calling, structured-outputs, vision
    Tools: web-search, file-search, image-generation, code-interpreter, mcp
    Knowledge cutoff: 2024-05-31
    Reasoning tokens supported
    Snapshots: o3, o3-2025-04-16
    Description: Well-rounded and powerful model across domains. Sets a new standard for math, science, coding, and visual reasoning tasks. Also excels at technical writing and instruction-following. Use it to think through multi-step problems that involve analysis across text, code, and images.
  [O4M] O4-mini (o4-mini)
    Provider: openai
    Context: 200,000 tokens (max output: 100,000 tokens)
    Capabilities: text, reasoning, analysis, coding, writing, visual-reasoning
    Modalities: text-input, text-output, image-input
    Pricing: $1.10/1M input, $4.40/1M output tokens
    Features: streaming, function-calling, structured-outputs, vision, fine-tuning
    Tools: web-search, file-search, code-interpreter, mcp
    Knowledge cutoff: 2024-05-31
    Reasoning tokens supported
    Snapshots: o4-mini, o4-mini-2025-04-16
    Description: Faster, more affordable reasoning model optimized for fast, effective reasoning with exceptionally efficient performance in coding and visual tasks.
%%> \model-catalog[sort="provider"]
Model Catalog (6 models):
Anthropic Models:
  [CO37] Claude 3.7 Opus (claude-3-7-opus-20240229)
    Provider: anthropic
    Context: 200,000 tokens (max output: 4,096 tokens)
    Capabilities: text, reasoning, analysis, coding, writing, math, research, visual-reasoning
    Modalities: text-input, text-output, image-input
    Pricing: $15.00/1M input, $75.00/1M output tokens
    Features: streaming, function-calling, structured-outputs, vision
    Knowledge cutoff: 2023-08-01
    Description: Top-level intelligence, fluency, and understanding
  [CS37] Claude 3.7 Sonnet (claude-3-7-sonnet-20250219)
    Provider: anthropic
    Context: 200,000 tokens (max output: 64,000 tokens)
    Capabilities: text, reasoning, analysis, coding, writing, math, visual-reasoning, extended-thinking
    Modalities: text-input, text-output, image-input
    Pricing: $3.00/1M input, $15.00/1M output tokens
    Features: streaming, function-calling, structured-outputs, vision
    Knowledge cutoff: 2024-11-01
    Description: High intelligence with toggleable extended thinking
  [CO4] Claude Opus 4 (claude-opus-4-20250514)
    Provider: anthropic
    Context: 200,000 tokens (max output: 32,000 tokens)
    Capabilities: text, reasoning, analysis, coding, writing, math, research, visual-reasoning, extended-thinking
    Modalities: text-input, text-output, image-input
    Pricing: $15.00/1M input, $75.00/1M output tokens
    Features: streaming, function-calling, structured-outputs, vision
    Knowledge cutoff: 2025-03-01
    Description: Highest level of intelligence and capability
  [CS4] Claude Sonnet 4 (claude-sonnet-4-20250514)
    Provider: anthropic
    Context: 200,000 tokens (max output: 64,000 tokens)
    Capabilities: text, reasoning, analysis, coding, writing, math, visual-reasoning, extended-thinking
    Modalities: text-input, text-output, image-input
    Pricing: $3.00/1M input, $15.00/1M output tokens
    Features: streaming, function-calling, structured-outputs, vision
    Knowledge cutoff: 2025-03-01
    Description: High intelligence and balanced performance
Openai Models:
  [O3] O3 (o3)
    Provider: openai
    Context: 200,000 tokens (max output: 100,000 tokens)
    Capabilities: text, reasoning, analysis, coding, writing, math, visual-reasoning, instruction-following, technical-writing
    Modalities: text-input, text-output, image-input
    Pricing: $2.00/1M input, $8.00/1M output tokens
    Features: streaming, function-calling, structured-outputs, vision
    Tools: web-search, file-search, image-generation, code-interpreter, mcp
    Knowledge cutoff: 2024-05-31
    Reasoning tokens supported
    Snapshots: o3, o3-2025-04-16
    Description: Well-rounded and powerful model across domains. Sets a new standard for math, science, coding, and visual reasoning tasks. Also excels at technical writing and instruction-following. Use it to think through multi-step problems that involve analysis across text, code, and images.
  [O4M] O4-mini (o4-mini)
    Provider: openai
    Context: 200,000 tokens (max output: 100,000 tokens)
    Capabilities: text, reasoning, analysis, coding, writing, visual-reasoning
    Modalities: text-input, text-output, image-input
    Pricing: $1.10/1M input, $4.40/1M output tokens
    Features: streaming, function-calling, structured-outputs, vision, fine-tuning
    Tools: web-search, file-search, code-interpreter, mcp
    Knowledge cutoff: 2024-05-31
    Reasoning tokens supported
    Snapshots: o4-mini, o4-mini-2025-04-16
    Description: Faster, more affordable reasoning model optimized for fast, effective reasoning with exceptionally efficient performance in coding and visual tasks.